### ISSUE TITLE
Fire an event before reloading

### ISSUE THREAD

**Issue by joankaradimov** [livereload/livereload-js#82]
The `LiveReload` object receives `LiveReloadConnect`/`LiveReloadDisconnect` events. Would it be possible/reasonable to trigger a `LiveReloadBeforeReload` event (or something similar) before the actual reloading is done? This will allow to distinguish between user initiated navigation and the reloading done by `livereload-js`.

**Comment by smhg** [comment]
You mean `document` receives these events, right? To avoid confusion.
Firing a `LiveReloadReload` event from a reload event [before the reloader](https://github.com/livereload/livereload-js/blob/046ebda8b6c2de17e1fcd4d867496a51506fd4b0/src/livereload.js#L124) sounds reasonable.

**Comment by joankaradimov** [comment]
> You mean `document` receives these events, right?

You are right, yes. The `LiveReload` object actually receives `connect`/`disconnect` and then fires `LiveReloadConnect`/`LiveReloadDisconnect` on the document.

> Firing a LiveReloadReload event from a reload event before the reloader sounds reasonable.

So adding this line in `performReload` would be ok:
```javascript
CustomEvents.fire(document, 'LiveReloadReload');
```

Should I open a pull request or can someone else implement this? Also - does this require a test?

**Comment by smhg** [comment]
A PR would definitely be appreciated, thanks!
There seem to be no mocha tests for this functionality. So let's add them, however basic.
